### ISSUE TITLE
PsychoPy Installation Script for Linux

### ISSUE THREAD

**Issue by wieluk** [psychopy/psychopy#6802]
I’ve developed a bash installation script for Linux that significantly improves upon the current installation process. My script addresses many of the missing packages and complexities that the existing script does not handle, making the setup much smoother and faster across various distributions.

You can find the installer on [GitHub](https://github.com/wieluk/psychopy_linux_installer). It has been thoroughly tested using GitHub Actions, with an automated list in the repository showing all the distributions where the script works or doesn't work.

Currently, the installer is compatible with the following distributions:

- Ubuntu 24.04, 22.04, 20.04
- Pop!_OS 22.04
- Debian 12, 11
- Fedora 40, 39
- Rocky Linux 9
- CentOS 9

The script handles the installation of Python, wxPython, and PsychoPy, managing versioning for virtual environments, adding PsychoPy to the PATH, and even creating a desktop shortcut. It also gives users the option to choose specific versions if they prefer.

I believe this is a major upgrade to the current installation script, and I would be happy to contribute it to the PsychoPy repository or work on integrating it.

To-dos:

- Reconsider the storage location for prebuilt wxPython and Python binaries.
- Conduct further testing.
- Get Pacman-based distributions like EndeavourOS or Manjaro working.
- Possibly refactor the script.
- Reduce complexitiy.

**Comment by peircej** [comment]
Hi @wieluk and thanks. This looks cool. A few of us have been discussing, over email, the strategies and the pros/cons around this stuff (like whether to use sh or py for the script, and whether to build a wheel or install a wheel). I can certainly see that your sh script has taken a number of steps forwards, like adding options for some of the things we were discussing

Regarding the issue with 2024.2.1 not working on linux, I'm puzzled what could cause that 🤔

**Comment by wieluk** [comment]
I believe using sh makes more sense in this context, especially since a new Python version often needs to be installed anyway. Most modern OS distributions are already using Python 3.11 or higher, so relying on Python for the installer adds an unnecessary layer of complexity.

Providing a pre-built wheel is a great time-saver, but having a fallback to build from source is crucial to ensure compatibility across different systems.

The GUI bug is indeed puzzling, and it's particularly strange that it resolves itself after installing an earlier version.

**Comment by peircej** [comment]
Yes, I wasn't thinking it would necessarily be the same python executable that we'd use for the app but the reasons I/we were thinking about py script to do the install were that 
1. psychopy contributors would all be familiar (but often less familiar with sh - for instance I personally have to check the syntax for sh much more often than for py)
2. this would also happily run on other platforms, so we could extend the concept to run on macos/windows

I think 1. is still true but for 2. maybe we just create a separate (py) script for installing on those. They don't need anything like the same complexity as linux installs (e.g. wxpython hassles) and the more relevant code might get lost there

But I'll pop you an email so we can chat about it in more detail with the other people that have been working on similar ideas

**Comment by yarikoptic** [comment]
I was about to file an issue that my attempt of getting a singularity container with usable psychopy has failed (builds/installs but running runs into dri libraries gotchas seems likely due to conflict between conda and system wide). Now I will try "the script"! ;-)  Thanks @wieluk

**Comment by yarikoptic** [comment]
FWIW, and since there was discussion on "let's do in Python".  As we collected a growing number of "howtos" across different ways git-annex and DataLad could have been deployed, we crafted a similar installer helper for DataLad and "related" (git-annex, miniconda, rclone, ...):  https://github.com/datalad/datalad-installer/ and it is written in Python (but pure Python no external libraries), and has quite modular organization etc happen someone decided to adopt infrastructure or add `psychopy` component in there.   We use that installer heavily across CI runs to test against different versions of git-annex, python etc, and also mention it as a solution in our handbook: https://handbook.datalad.org/en/latest/intro/installation.html

**Comment by wieluk** [comment]
The project is now stable, and the script seems to be working well. It still leverages cloud.uni-graz.at for quick installs, with a fallback to the original source if needed. My local GitHub runners handle testing and uploading both wheels and precompiled Python versions. If there's interest, I could start considering an implementation. Let me know if you'd like to explore this further.
